### ISSUE TITLE
automate processes like updating documentation, style, etc?

### ISSUE THREAD

**Issue by k-doering-NOAA** [r4ss/r4ss#516]
I see there is a way to allow [github actions to push commits to a repository](https://github.com/actions/checkout#push-a-commit-using-the-built-in-token). I wonder if it would be worth it to try to automate some common workflows,?

For example, I see in the tidyr package, there is a [workflows setup](https://github.com/tidyverse/tidyr/blob/master/.github/workflows/pr-commands.yaml) that look for a certain issue comment on pull requests, and if found, will [use styler to style the pr automatically](https://github.com/tidyverse/tidyr/pull/1110#issuecomment-814769637) . This seems helpful if all new development is occurring through pull requests.

I think another option would just be to run devtools::document() and perhaps styler and the rm_dollar_sign function on a schedule (perhaps weekly?). I imagine we could do this directly on the main branch, or maybe have a branch made and submitted as a PR to review, if we want to review the differences just in case...

Thoughts if this would be useful/worth pursuing?

**Comment by iantaylor-NOAA** [comment]
I think automatically running `devtools::document()` on pull requests seems very reasonable. On the other hand, I worry that `r4ss::rm_dollar_sign()` and `styler::style_pkg()`, while really valuable, could cause confusion if someone creates a PR, and then further changes are suggested before merging, but the code has changed from what got committed due to the changes those function make. So a weekly schedule, or an option to just run an an action on demand, might be the better approach. But I'm also a little confused by the "look for a certain issue comment" part. If that's something that only routine contributors to r4ss would add, fully understanding it's impact, then I'm good with all 3 steps being part of the process for every PR.

**Comment by k-doering-NOAA** [comment]
Thanks @iantaylor-NOAA !

> On the other hand, I worry that r4ss::rm_dollar_sign() and styler::style_pkg(), while really valuable, could cause confusion if someone creates a PR, and then further changes are suggested before merging, but the code has changed from what got committed due to the changes those function make. So a weekly schedule, or an option to just run an an action on demand, might be the better approach.

Absolutely, it would be confusing!

>  But I'm also a little confused by the "look for a certain issue comment" part. If that's something that only routine contributors to r4ss would add, fully understanding it's impact, then I'm good with all 3 steps being part of the process for every PR.

I'm not sure if there are controls over who can trigger the action and I agree it doesn't really make sense to allow anyone to trigger it whenever they want on a PR.

Based on your comments, I think it makes sense to set up (when there is time to do so) something to run devtools::document() since it only impacts files that folks aren't directly editing and having up-to-date documentation consistent with the codebase would make a positive impact for r4ss users. Running `r4ss::rm_dollar_sign()` and `styler::style_pkg()` more often might do more harm than good, because although the code would have more consistent style and could be easier to read, having large style changes frequently mght lead to more confusion when trying to merge in work.

**Comment by kellijohnson-NOAA** [comment]
I fully support automating document()!

**Comment by iantaylor-NOAA** [comment]
@k-doering-NOAA, this issue was closed because the PR #570 included the text "Addresses #516" in the commit summary (more detail [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) on this feature which I just learned by seeing @kellijohnson-NOAA use it). But it's probably better to leave this open until we've refined the process a bit.

The first automated Pull Request has been made (https://github.com/r4ss/r4ss/pull/572). The changes look good but I'm not seeing any progress on the automatic checks, so am holding off on merging it.

**Comment by k-doering-NOAA** [comment]
Yes, I noticed for some reason the automatic checks aren't triggered by the PR for style and document; I'm not sure why or how to make them start!  Since they are just styling changes, I hope we can just merge into main to run the checks? Definitely would be nicer to run before merging in, though...

I closed the first auto PR, because I had made some modifications to the job (tested already with r4ss). I restarted the job manually, which will probably open another PR.
